### PR TITLE
[#44] [Backend] As a user, I can get the result data of multi keywords from Google Search

### LINKED ISSUE #44: [Backend] As a user, I can get the result data of multi keywords from Google Search
## Acceptance Criteria

- Setup worker using [work](https://github.com/gocraft/work)
- Queue Keyword crawling in the background

### PR DESCRIPTION
**Pull request by markgravity**
- Close #44 

## What happened 👀

- Setup worker to run jobs in the background
- Setup Redis to serve enqueue jobs
- Update `deploy.yml` to deploy worker
- Add `Crawl` job

## Insight 📝

n/a

## Proof Of Work 📹

https://github.com/markgravity/golang-ic/assets/17875522/ba175aca-890d-4850-a928-6c805f27732c

### CODE REVIEW

File: lib/services/crawler/crawler.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ var selectors = map[string][]string{
 const urlPattern = "https://www.google.com/search?q=%s"
 
 func (c *Crawler) Run() error {
-	if c.Keyword == nil {
+	if c.Keyword == nil || c.Keyword.Base.ID.ID() == 0 {
```
Comment (malparty) [inline lib/services/crawler/crawler.go]:
```suggestion
	if c.Keyword == nil || c.Keyword.Base.ID == uuid.NullUUID {
```

Or something similar. Let's not convert the UUID to integers in order to assert if it is null :)

Comment (markgravity) [reply lib/services/crawler/crawler.go]: @malparty it doesn't work 🙏 
CleanShot 2023-08-31 at 13 20 28

Comment (suho) [reply lib/services/crawler/crawler.go]: Maybe this 
```suggestion
	if c.Keyword == nil || c.Keyword.Base.ID.ID() == uuid.Nil {
```

Comment (markgravity) [reply lib/services/crawler/crawler.go]: @suho @malparty it's perfect, I updated it here [c59348a](https://github.com/markgravity/golang-ic/pull/47/commits/c59348a3a013a45a383d8264d0708be0603d55fb)

**Review by malparty — APPROVED** [review]

**Review by suho — APPROVED** [review]

**Pull request merged**